test: Mitigate media block test flakiness

dcalhoun opened the pull request:
Relates to https://github.com/wordpress-mobile/gutenberg-mobile/issues/6233. Partially reverts https://github.com/wordpress-mobile/WordPress-iOS/pull/21582. 

Mitigate media block test flakiness by pasting URLs rather than typing them. The
rapid rate at which XCTests type resulted in dropped URL characters leading to
invalid media URLs. This is undoubtedly a performance issue that could be
addressed directly, but pasting the URL feels like an appropriate solution for
resolving the test issue for now.

To test: ensure the CI tasks succeed.

## Regression Notes
1. Potential unintended areas of impact
    Other automated tests could begin failing.
2. What I did to test those areas of impact (or what existing automated tests I relied on)
    Verifying CI tasks succeed.
3. What automated tests I added (or what prevented me from doing so)
    Improved the robustness of media block tests.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

dcalhoun committed:
test: Paste media URL to avoid dropped characters during typing

The rapid typing within XCTest occasionally results in dropped
characters in the "Insert from URL" text field. This means a valid URL
of `https://wordpress.org/video.mp4` ends up captured as
`h://wordpress.org/video.mp4`.

Addressing the underlying performance issue might be ideal, but pasting
the entire URL instead of typing is a reasonable workaround for now.

dcalhoun committed:
test: Increase media block predicate specificity

The previous predicates passed after merely adding a block, as it
asserted aspects that do not verify the act of attaching media. The new
predicates are only present if attaching media to the blocks succeeds.
This increases the coverage for the tests.

dcalhoun committed:
test: Re-enable media block test for iPhone devices

The other changes in this branch appear to resolve the issues causing
iPhone failures.

jostnes commented on WordPress/UITestsFoundation/Screens/Editor/BlockEditorScreen.swift:
```diff
@@ -186,25 +186,16 @@ public class BlockEditorScreen: ScreenObject {
-        let imagePredicate = NSPredicate(format: "label == 'Image Block. Row 1'")
-        let videoPredicate = NSPredicate(format: "label == 'Video Block. Row 2'")
-        let audioPredicate = NSPredicate(format: "label == 'Audio Block. Row 3'")
+        let imagePredicate = NSPredicate(format: "label == 'Image caption. Empty'")
+        let videoPredicate = NSPredicate(format: "label == 'Video caption. Empty'")
+        let audioPredicate = NSPredicate(format: "label == 'Audio Player'")
```
Any particular reason this was updated? In the previous version the Row number was added so we'd know that the blocks are added and will maintain in the order that they were added.

dcalhoun replied:
👋🏻 @jostnes. Thanks for the review! 

I expanded upon the reason for these changes in the relevant commit message: https://github.com/wordpress-mobile/WordPress-iOS/pull/21639/commits/aece7ee7b229c994d69835fb686ea268fa227d09. Basically, the prior assertions passed after merely _inserting the blocks_. Whether or not attaching media succeeded had no impact on whether the test case was deemed successful. I.e. a block could be inserted and attaching media could fail, but the test would still pass. 

As you pointed out, the new assertions no longer cover the block order, but I perceived that to be less of a focus for the test at the time. If that is important, we could expand the assertions to cover both ordering and media attachment or revert my changes. 

WDYT?

jostnes replied:
Thanks for explaining the reason (and sorry I missed the commit message earlier 🙏), I checked and you're right, though I think it would be good to get the test to verify the order too. I would vote for the assertions to cover both ordering and media attachments.

dcalhoun replied:
Thanks for the feedback. I addressed this https://github.com/wordpress-mobile/WordPress-iOS/pull/21639/commits/d0a0f2ab026134e46995de9b38cbc7d6a1c6c735.

jostnes reviewed (commented):
Thanks for updating the test! And good catch finding out the reason of failure. Changes related to adding URL looks good and CI is 🟢 but I added one question related to the updated assertion that I'd like to understand before approving. Thanks!

dcalhoun committed:
test: Assert media block position and attachments

Asserting the block position in addition to the media attachment
increases the robustness of the test.

jostnes reviewed (approved):
Thank you for considering the feedback and making the changes! :shipit:
